### ISSUE TITLE
Come up with a better documentation format (and use it)

### ISSUE THREAD

**Issue by sdegutis** [sdegutis/mjolnir#18]
- [x] show return values
- [x] document arg types and return types
- [x] make the easy stuff easy to read and the hard stuff possible to read without being ugly
- [ ] maybe find a way to make it prettier?
- [x] use the wiki, don't use github-pages or anything like that

**Comment by sdegutis** [comment]
1. Return values are specified by `->` rather than using the C style.
2. Include type in parameter names, rather than specifying types and parameters separately.
3. When you just return self, explicitly say that.
4. Arrays use the postfix brackets, i.e. `window[]`.
5. Function parameters should follow all these rules and include parentheses.
6. Instance-overridable functions should be an assignment to a regular function.
7. One-off table parameter types can show key names inline, i.e. `{name, age}`.
8. Use ellipses to indicate variadic parameters.
9. Specify default arguments with an equal sign inline.
10. Show non-function values by assignment.

``` lua
api.geometry.hypot(point) -> int
api.geometry.intersectionrect(rect1, rect2) -> rect
api.hotkey:enable() -> self
api.hotkey:enable() -> self
api.hotkey.keys = {}

api.pathwatcher(path, fn) -> pathwatcher
api.pathwatcher:start()
api.pathwatcher:stop()

api.screen.allscreens() -> screen[]
api.screen.mainscreen() -> screen
api.screen:frame_including_dock_and_menu() -> rect

-- function parameter
api.timer.runonce(fn())

-- instance-overridable function
api.textgrid.resized = function()
api.textgrid.keydown = function({cmd, ctrl, alt, key})

api.hotkey(mods, key, fn()) -- need comments to specify mods and key

api.call(fn(), ...)
api.errorhandler = function(err)

api.geometry.rotateccw(point, aroundpoint, ntimes = 1)
```

**Comment by sdegutis** [comment]
1. Each function or value should fall under a single module.
2. Each module (i.e. `api.textgrid` and `api.log`) should have its own documentation.
3. Every function or value in a module should have (a) a definition string, and (b) a documentation string.

**Comment by sdegutis** [comment]
Even though this issue is closed, it should be taken into account in #24.
